lib/smart_proxy_salt/cli.rb

```diff
@@ -62,54 +62,42 @@ def autosign_list
         end
 
         def append_value_to_file(filepath, value)
```

Comment:
I still have my doubts about this file and concurrency, but that was already a problem prior to this so we can consider it out of scope. However, it is important to realize that the Smart Proxy can have multiple threads and I wonder if this can result in corrupted files.

`remove_value_from_file` probably wouldn't have corrupted files, but can also have race conditions.

Reply:
Do you suggest implementing a mutex for these two functions? I thought (without further research) the whole `File.open` block may be threadsafe as long as the same file is opened, but the only information I found on it is rather old and may not be up to date (see the [forum](https://www.ruby-forum.com/t/multithreaded-file-access/50995/11)).
Other information is given [here](https://stackoverflow.com/a/23334292) which implies that we could use `File.open(path).flock(File::LOCK_EX)` and the unlock equivalent `File.open(path).flock(File::LOCK_UN)`.

What do you think about it?

Reply:
Looks like I forgot to write it down, but perhaps it should be fixed in a separate issue/PR. This PR already improves things by a lot and the concurrency problem (if there is any) was there before this PR. So I think we can now merge this.